File: LeetcodeProblems/blind-75/interval/merge-intervals/solution.py
# Merge Intervals - LeetCode #56

from typing import List

class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if not intervals:
            return []
        
        intervals.sort(key=lambda x: x[0])
        
        result = [intervals[0]]
        
        for interval in intervals:
            current = result[-1]
            if interval[0] <= current[1]:
                current[1] = max(current[1], interval[1])
            else:
                result.append(interval)
        
        return result

File: LeetcodeProblems/blind-75/interval/merge-intervals/solution.js
// Merge Intervals - LeetCode #56

/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    if (intervals.length === 0) return [];
    
    intervals.sort((a, b) => a[0] - b[0]);
    
    const result = [intervals[0]];
    
    for (let interval of intervals) {
        const current = result[result.length - 1];
        if (interval[0] <= current[1]) {
            current[1] = Math.max(current[1], interval[1]);
        } else {
            result.push(interval);
        }
    }
    
    return result;
};
